### PR TITLE
[T-296] mdx transformer

### PR DESCRIPTION
**Pull request by DeMoorJasper**
# ↪️ Pull Request

A simple MDX transformer based on #3262

## ✔️ PR Todo

- [x] Added/updated unit tests for this change
- [x] Filled out test instructions (In case there aren't any unit tests)
- [x] Included links to related issues/PRs

### CODE REVIEW

**Comment by parcel-benchmark** [comment]
## Benchmark Results
<details><summary>packages/benchmarks/kitchen-sink 🚨</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | **FAILED** | -0.00ms |
| Cached | **FAILED** | -0.00ms |

#### Cold Bundles

*No bundles found, this is probably a failed build...*
#### Cached Bundles

*No bundles found, this is probably a failed build...*
</p></details>

<details><summary>packages/benchmarks/react-hn 🚨</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | **FAILED** | -0.00ms |
| Cached | **FAILED** | -0.00ms |

#### Cold Bundles

*No bundles found, this is probably a failed build...*
#### Cached Bundles

*No bundles found, this is probably a failed build...*
</p></details>

<details><summary>packages/benchmarks/ak-editor 🚨</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | **FAILED** | -0.00ms |
| Cached | **FAILED** | -0.00ms |

#### Cold Bundles

*No bundles found, this is probably a failed build...*
#### Cached Bundles

*No bundles found, this is probably a failed build...*
</p></details>

**[Click here to view a detailed benchmark overview.](https://benchmark.parceljs.org/benchmark/254386324624114185)**

File: packages/configs/default/index.json [hunk 1]
```diff
@@ -25,6 +25,7 @@
     ],
     "*.pug": ["@parcel/transformer-pug"],
     "*.coffee": ["@parcel/transformer-coffeescript"],
+    "*.{mdx}": ["@parcel/transformer-mdx"],
```
Comment (mischnic) [inline packages/configs/default/index.json]:
```suggestion
    "*.mdx": ["@parcel/transformer-mdx"],
```

File: packages/core/integration-tests/test/mdx.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+const assert = require('assert');
+const path = require('path');
+const {bundle, run} = require('@parcel/test-utils');
+
+describe('mdx', function() {
+  it('should support bundling MDX', async function() {
+    let b = await bundle(path.join(__dirname, '/integration/mdx/index.mdx'), {
+      patchConsole: false,
```
Comment (wbinnssmith) [inline packages/core/integration-tests/test/mdx.js]: is this necessary or is it debugging?

Comment (DeMoorJasper) [reply packages/core/integration-tests/test/mdx.js]: debugging

**Pull request merged**